src/control/server/ctl_ranks_rpc.go

```diff
@@ -315,7 +315,13 @@ func (svc *ControlService) ResetFormatRanks(ctx context.Context, req *ctlpb.Rank
 	}
 
 	// ignore poll results as we gather state immediately after
-	if _, err = pollInstanceState(ctx, instances, (*EngineInstance).isAwaitingFormat,
+	if _, err = pollInstanceState(ctx, instances, func(e Engine) bool {
+		ei, ok := e.(*EngineInstance)
+		if !ok {
+			return false
+		}
+		return ei.isAwaitingFormat()
```

Comment:
I wonder, is there a reason not to make this method part of the interface too?

Reply:
All right, I'm interested to see how this takes shape. From my view this seems like useful information to expose to callers, since it conveys something pretty important about the engine state.

Reply:
I've added it to the list of questionable interface methods for now.